### ISSUE TITLE
CombinedIndexBuildItem to receive ClassInfo on inherited interfaces

### ISSUE THREAD

**Issue by oztimpower** [quarkusio/quarkus#7955]
**Description**
For example, per Amazon Lambda extension documentation updates:

> Writing Skills with the Amazon Alexa ASK SDK v2
> 
>  The starting point for an Alexa Skill on AWS Lambda is typically via subclassing `SkillStreamHandler`, which implements the `RequestStreamHandler::requestHandler` method.  If you examine the source code for this class it is very simple, and can be implemented directly without this helper base class.
> 
>  NOTE: Quarkus requires a direct implementation of `RequestStreamHandler`, as the build process will scan only directly implemented interfaces, inherited interfaces will not be seen.

**Implementation ideas**

Generated ClassInfo, as part of the CombinedIndexBuildItem, to include all inherited interfaces. It currently only contains directly implemented interfaces.

For example in Amazon Lambda, AmazonLambdaProcesser, the code below looking for a specific interface implementation, would see the inherited interfaces.

`      Collection<ClassInfo> allKnownImplementors = combinedIndexBuildItem.getIndex().getAllKnownImplementors(REQUEST_HANDLER);
`
```
    @BuildStep
    List<AmazonLambdaBuildItem> discover(CombinedIndexBuildItem combinedIndexBuildItem,
            Optional<ProvidedAmazonLambdaHandlerBuildItem> providedLambda,
            BuildProducer<AdditionalBeanBuildItem> additionalBeanBuildItemBuildProducer,
            BuildProducer<ReflectiveHierarchyBuildItem> reflectiveHierarchy,
            BuildProducer<ReflectiveClassBuildItem> reflectiveClassBuildItemBuildProducer) throws BuildException {
  
      Collection<ClassInfo> allKnownImplementors = combinedIndexBuildItem.getIndex().getAllKnownImplementors(REQUEST_HANDLER);

      Collection<ClassInfo> allKnownStreamImplementors = combinedIndexBuildItem.getIndex()
                .getAllKnownImplementors(REQUEST_STREAM_HANDLER);


```
/cc @gsmet 
/cc @patriot1burke

**Comment by geoand** [comment]
I think this request is essentially this issue: https://github.com/wildfly/jandex/issues/65

**Comment by gsmet** [comment]
For now, I would just also get the classes implementing `SkillStreamHandler` and merge them with the others.

Not ideal and we should definitely pursue this on the Jandex side but good enough for this particular case.

**Comment by oztimpower** [comment]
Closing as a workaround has been implemented in #7985, and as above there is already a FR open on wildfly/jandex#65
